puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tarballsRelease.groovy

```diff
@@ -0,0 +1,105 @@
+    def major_version = env.getProperty('major_version')
+    def ruby_ver = '2.4.0'
+
+    stages {
+        stage('Parallel') {
+
+            parallel {
+                stage('Foreman') {
+                    steps {
+                        verify_tag('foreman', version)
+                        build_tarball('foreman', version, ruby_ver)
+                    }
+                }
+                stage('Smart Proxy') {
+                    steps {
+                        verify_tag('smart-proxy', version)
+                        build_tarball('smart-proxy', version, ruby_ver)
+                    }
+                }
+                stage('Foreman SELinux') {
+                    steps {
+                        verify_tag('foreman-selinux', version)
+                        build_tarball('foreman-selinux', version, ruby_ver)
+                    }
+                }
+                stage('Foreman Installer') {
+                    steps {
+                        verify_tag('foreman-installer', version)
+                        build_tarball('foreman-installer', version, ruby_ver)
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            build job: 'release_packages', parameters: [
+                [$class: 'StringParameterValue', name: 'version', value: version],
+                [$class: 'StringParameterValue', name: 'major_version', value: major_version]
+            ]
+        }
+    }
+}
+
+void verify_tag(project, version) {
+    dir(project) {
+        deleteDir()
+        git url: "https://github.com/theforeman/${project}", branch: 'develop'
+        sh """
+        if [ \"$(git tag -l ${version} | wc -l)\" -ne 1 ]; then
+            echo \"ERROR: cannot find tag ${version}\"
+            git tag -l
+            exit 1
+        fi
+        exit 0
+        """
+        deleteDir()
+    }
+}
+
+void build_tarball(project, version, ruby_ver) {
+    def base_dir = "/var/www/vhosts/downloads/htdocs/foreman"
+    def rake = "rake"
+
+    dir(project) {
+        deleteDir()
+        checkout scm: [$class: 'GitSCM',
+            userRemoteConfigs: [[url: "git@github.com:theforeman/${project}.git"]],
+            branches: [[name: "refs/tags/${version}"]]], changelog: false, poll: false
+
+        configureRVM(ruby_ver)
+
+        if (project == 'foreman') {
+            sh "cat config/settings.yaml.example > config/settings.yaml"
+            sh "cat config/database.yml.example > config/database.yml"
+        }
+
+        env.setProperty('DEBUG_RESOLVER', '1')
+
+        if (fileExists("Gemfile")) {
+            withRVM(["bundle install --without=development --retry=5"], ruby_ver)
+            rake = "bundle exec rake"
+        }
+
+        withRVM(["${rake} pkg:generate_source", ruby_ver)
+
+        sh "ssh root@theforeman.org \"mkdir -p ${base_dir}/\" || true"
+        sh "rsync -v --ignore-existing pkg/* root@theforeman.org:${base_dir}/"
```

Comment:
This might seem like an early optimization, but we have nightly jobs that build tarballs for example to then build nightly RPMs that could take use of this function.

Reply:
Do you know of any jobs off the top of your head?

Reply:
I don't think this suggestion is a bad idea. In fact I think it's a very good idea, but maybe outside of the scope of this PR since it's been open for some time now.

Reply:
ACK